src/seraphis_impl/checkpoint_cache.h

```diff
@@ -113,14 +128,14 @@ class CheckpointCache
 
 //member variables
     /// minimum checkpoint index
-    const std::uint64_t m_min_checkpoint_index;
+    std::uint64_t m_min_checkpoint_index;
 
     /// config
-    const CheckpointCacheConfig m_config;
-    static const std::uint64_t m_window_size{3};
+    CheckpointCacheConfig m_config;
+    std::uint64_t m_window_size{3};
 
     /// stored checkpoints
-    std::map<std::uint64_t, rct::key> m_checkpoints;
+    serializable_map<std::uint64_t, rct::key> m_checkpoints;
```

Comment:
Well, I am not the best person to ask, because I myself as the "master" over the Seraphis library probably wouldn't go through all the complications with those `ser_` classes and add serialization instrumentation directly to the Seraphis library struct and classes ...

But now, as we do separate, I think we could well go the extra mile and keep *any* serialization stuff out of the Seraphis library, as a matter of principle, and for not mixing things.

Maybe one day somebody will want to serialize Seraphis library structures with a completely different technology than `binary_archive` where, at least in theory, such a `serializable_map` would be not helpful in a good case and a problem with some bad luck.